tools/run.py

```diff
@@ -24,11 +24,14 @@ def register(command):
         default=False,
         help="Run the specified program using valgrind",
     )
+    command.add_argument(
+        "--dont-append-bin", dest="use_bin", action="store_false", default=True, help="(Advanced) run a shell command"
```

Comment:
You're prepending `bin`, not appending. Also, I think this would be better if it were something like `run-command` or `run-program` or `execute-command` or something, rather than the logically negative, implementation-specific `dont-prepend-bin`.
Also, please add a short version to this argument.